extract/eventlog/eventlog_test.go

```diff
@@ -0,0 +1,242 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package eventlog
+
+import (
+	"os"
+	"path"
+	"testing"
+
+	"github.com/google/gce-tcb-verifier/eventlog"
+	"github.com/google/gce-tcb-verifier/testing/match"
+	"github.com/google/go-cmp/cmp"
+	test "github.com/google/go-sev-guest/testing"
+)
+
+const (
+	myGUID  = "6a7b6885-92bc-40cd-9fb5-300f9d1eb0ed"
+	rimGUID = "c51b6d7f-9c2a-42d6-be47-ca1368bdc333"
+)
+
+var myEfiGUID = []byte{0x85, 0x68, 0x7b, 0x6a, 0xbc, 0x92, 0xcd, 0x40, 0x9f, 0xb5, 0x30, 0x0f, 0x9d, 0x1e, 0xb0, 0xed}
+
+func TestUEFIVariable(t *testing.T) {
+	tcs := []struct {
+		name    string
+		data    []byte
+		want    string
+		wantErr string
+	}{
+		{
+			name: "happy path",
+			data: append(myEfiGUID, 'V', 0, 'a', 0, 'r', 0, 0, 0),
+			want: "Var-" + myGUID,
+		},
+		{
+			name:    "odd length",
+			data:    append(myEfiGUID, 'V', 0, 'a', 0, 'r', 0, 0),
+			wantErr: "couldn't read variable name as UTF-16 string",
+		},
+		{
+			name:    "bad terminator no zeros",
+			data:    append(myEfiGUID, 'V', 0, 'a', 0, 'r', 0),
+			wantErr: "00-terminated UTF-16 string",
+		},
+		{
+			name:    "too short",
+			data:    []byte{0, 1},
+			wantErr: "too short: 2 bytes",
+		},
```

Comment:
Can you add some tests that test with non-ascii strings, including some that involve utf-16 surrogate pairs?